front/src/modules/pathfinding/utils.ts

```diff
@@ -33,9 +36,38 @@ export const formatSuggestedOperationalPoints = (
     offsetOnTrack: op.part.position,
     track: op.part.track,
     positionOnPath: op.position,
-    coordinates: getPointCoordinates(geometry, pathLength, op.position),
+    coordinates:
+      (geometry !== undefined &&
+        pathLength !== undefined &&
+        getPointCoordinates(geometry, pathLength, op.position)) ||
+      undefined,
   }));
 
+export const addTrackNamesToSuggestedOperationalPoints = async (
+  operationalPoints: SuggestedOP[],
+  getTrackSectionsByIds: (requestedTrackIds: string[]) => Promise<InfraObjectWithGeometry[]>
```

Comment:
Hum I don't think it's very useful to give the argument `getTrackSectionsById`, it would be better to call it outside `addTrackNamesToSuggestedOperationalPoints` and to give in argument only `operationalPoints` and `trackSections`.

Reply:
Sure!

Reply:
It does mean that every caller will have to perform the 3 same steps instead of just calling the function though :
-> extract ids to fetch
-> fetch the ids
-> call this function (which is reduced to its last step)

Plus if we need to make it generic, we may as well delete the function and just inline everything

Reply:
Yes you're right. I find the code way cleaner and easier to understand, like this, thanks :star_struck: